### PR TITLE
Lecture 6 -HW

### PR DESCRIPTION
**Pull request by IrinaZhernosek**
all tasks.

### CODE REVIEW

File: IrConsoleApp/Lecture6/Student.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs]: remove unused namespaces

File: IrConsoleApp/Lecture6/Student.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace IrynaZharnasek.Lecture6
+{
+    public class Student : Person
+    {
+        private string position;
+
+        public string Position
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs]: you may use auto property here

File: IrConsoleApp/Lecture6/Student.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        public Student(int age, string name = "underfined", string surname = "underfined", string position = "underfined"): base(age, name, surname)
+        {
+            this.position = position;
+        }
+
+        public Student(string name = "underfined", string surname = "underfined", string position = "underfined") : base(name, surname)
+        {
+            this.position = position;
+        }
+
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs:22]: between method should be only one empty line

File: IrConsoleApp/Lecture6/Student.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+
+        public void showStudentInfo() => Console.WriteLine($"name: {Name} surname:{Surname} age:{Age} position: {position}");
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs]: Name of method should start from upper case

File: IrConsoleApp/Lecture6/Student.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs]: remove empty lines after methods

File: IrConsoleApp/Lecture6/Person.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                if (value < 1 || value > 70)
+                    Console.WriteLine("The age should be between '0' and '71'");
+                else
+                    this.age = value;
+            }
+            get { return this.age; }
+        }
+
+        //defining access metod for name:
+        public string Name
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Person.cs]: you may use autopropety here as logic described in get/set blocks is kind of default

File: IrConsoleApp/Lecture6/Person.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            set
+            {
+                this.name = value;
+            }
+            get { return this.name; }
+        }
+
+        //defining access metod for surname:
+        public string Surname
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Person.cs]: you may use autopropety here as logic described in get/set blocks is kind of default

File: IrConsoleApp/Lecture6/Person.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+        }
+
+        //creating object of Class Student through Constructor by default:
+        public Person()
+        {
+
+        }
+
+        public void showPersonInfo() => Console.WriteLine($"name: {name} surname:{surname} age:{Age}");
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Person.cs]: Method name should be upper case

File: IrConsoleApp/Lecture6/Course.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Course.cs]: Please use ctrl K + D combination to format this and all files

Comment (ashurakova) [inline IrConsoleApp/Lecture6/Course.cs]: Please install Trailing whitespace extension and format all files

File: IrConsoleApp/Lecture6/Person.cs [hunk 10]
```diff
@@ -0,0 +1,82 @@
+﻿using IrynaZharnasek.Lecture6;
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Person.cs:1]: Remove unused namespace

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 11]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: Remove unused namespace

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IrynaZharnasek.Lecture6
+{
+    public class Mentor:Person
+    {
+        private string qualification;
+
+        public string Qualification
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: Can be done as autoproperty

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            set
+            {
+                this.qualification = value;
+            }
+            get { return this.qualification; }
+        }
+
+        //creating Mentor through Constructor by default:
+        public Mentor() { }
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: put {} into a new lines
like Mentor()
{
}

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+        //creating Mentor through Constructor (all parameters):
+        public Mentor(int age, string name = "underfined", string surname = "underfined", string qualification = "underfined") : base(age, name, surname)
+        {
+            this.qualification = qualification;
+        }
+
+        public void showMentorInfo() => Console.WriteLine($"name: {Name} surname:{Surname} age:{Age} qualification: {qualification}");
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: Method name should be in upper case

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: Remove empty line

File: IrConsoleApp/Lecture6/Course.cs [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+using IrynaZharnasek.Lecture6;
+
+namespace IrynaZharnasek.Lecture6
+{
+    public class Course
+    {
+        private string course;
+
+        private Student[] arrayOfStudents = new Student[100];
+        private Mentor[] arrayOfMentors = new Mentor[100];
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Course.cs]: As far as I understand, the current logic is the following: one course can have 100 Mentors. But I would say it's better the logic be the following: one course can have one Mentor. But Courses Library can have 100 mentors. And mentor should be assigned to the course via the courses library

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 14 above)
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: You may use overriding with this method as similar methods in Person and Student classes

File: IrConsoleApp/Lecture6/Student.cs (hunk 1 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 2 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 3 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs:22]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 4 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 5 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs]: Done

File: IrConsoleApp/Lecture6/Person.cs (hunk 6 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Person.cs]: Done

File: IrConsoleApp/Lecture6/Person.cs (hunk 7 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Person.cs]: Done

File: IrConsoleApp/Lecture6/Person.cs (hunk 8 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Person.cs]: Done

File: IrConsoleApp/Lecture6/Course.cs (hunk 9 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Course.cs]: Done

Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Course.cs]: Done

File: IrConsoleApp/Lecture6/Person.cs (hunk 10 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Person.cs:1]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 11 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 12 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 14 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 15 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 13 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Course.cs (hunk 16 above)
Comment (ashurakova) [reply IrConsoleApp/Lecture6/Course.cs]: still partly relevant:
1 course has has 10 cources
Example from real life - your c# course has inside some more courses - cooking course, photography course, UI/UX course.

Course should not have array of courses, courses library should have array of courses

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 14 above)
Comment (ashurakova) [reply IrConsoleApp/Lecture6/Mentor.cs]: not fixed

File: IrConsoleApp/Lecture6/Course.cs [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            for (var i = 0; i < arrayOfMentors.Length; i++)
+            {
+                if (arrayOfMentors[i] == null)
+                {
+                    arrayOfMentors[i] = mentor;
+                    break;
+                }
+            }
+        }
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Course.cs]: add empty line here
empty line should be before/after method

File: IrConsoleApp/Lecture6/CoursesLibrary.cs [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+            for (var i = 0; i < _allCourses.Length; i++)
+            {
+                if (_allCourses[i].CourseTitle == title)
+                {
+                    return _allCourses[i];
+                }
+            }
+
+            return null;
+        }
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/CoursesLibrary.cs:55]: add empty line

File: IrConsoleApp/Lecture6/CoursesLibrary.cs [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+            {
+                Course testing = GetCourseByTitle("Functional testing");
+                testing.AddNewStudent(student);
+                Console.WriteLine($"Student {student.Name} assighed to {testing.CourseTitle}");
+            }
+            else
+            {
+                Console.WriteLine($"No courses were found for {student.Name} position!");
+            }
+        }
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/CoursesLibrary.cs:75]: add empty line

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 20]
```diff
@@ -0,0 +1,21 @@
+﻿namespace IrynaZharnasek.Lecture6
+{
+    public class Mentor:Person
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: please format it by ctrl,k+D

File: IrConsoleApp/Lecture6/Mentor.cs [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+        }
+
+        //creating Mentor through Constructor (all parameters):
+        public Mentor(int age, string Name = "underfined", string Surname = "underfined", string qualification = "underfined") : base(age, Name, Surname)
+        {
+            Qualification = qualification;
+        }
+
+        public void ShowMentorInfo() => Console.WriteLine($"name: {Name} surname:{Surname} age:{Age} qualification: {Qualification}");
+
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Mentor.cs]: remove empty line (no need to put an empty line after method if it's the end of the plass)

File: IrConsoleApp/Lecture6/Person.cs [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace IrynaZharnasek.Lecture6
+{
+    public class Person
+    {
+        private int age;
+        //defining access metod for Student's Age:
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Person.cs:9]: add empty line before comments (here and below)

File: IrConsoleApp/Lecture6/Student.cs [hunk 23]
```diff
@@ -0,0 +1,22 @@
+﻿namespace IrynaZharnasek.Lecture6
+{
+    public class Student : Person
+    {
+        public string Position { get; set; }
+        //creating Student through Constructor by default:
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs:7]: add empty line before comment

File: IrConsoleApp/Lecture6/Student.cs [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+        public Student() {}
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Student.cs]: put {} into a new line

File: IrConsoleApp/Lecture6/Course.cs [hunk 25]
```diff
@@ -0,0 +1,78 @@
+﻿namespace IrynaZharnasek.Lecture6
+{
+    public class Course
+    {
+        private string course;
+        private Mentor mentor;
```
Comment (ashurakova) [inline IrConsoleApp/Lecture6/Course.cs:6]: mentor is unused, you need to sync logic: if you expected that 1 course should have 1 mentor, that you need to keep this variable and remove variable arrayOfMentors. If you expected that 1 course should have several mentors, that you need to remove mentor variable and keep arrayOfMentors, up to you

File: IrConsoleApp/Lecture6/Course.cs (hunk 16 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Course.cs]: Done

File: IrConsoleApp/Lecture6/Course.cs (hunk 17 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Course.cs]: Done

File: IrConsoleApp/Lecture6/CoursesLibrary.cs (hunk 18 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/CoursesLibrary.cs:55]: Done

File: IrConsoleApp/Lecture6/CoursesLibrary.cs (hunk 19 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/CoursesLibrary.cs:75]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 20 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 21 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

File: IrConsoleApp/Lecture6/Person.cs (hunk 22 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Person.cs:9]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 23 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs:7]: Done

File: IrConsoleApp/Lecture6/Student.cs (hunk 24 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Student.cs]: Done

File: IrConsoleApp/Lecture6/Course.cs (hunk 25 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Course.cs:6]: Done

File: IrConsoleApp/Lecture6/Mentor.cs (hunk 14 above)
Comment (IrinaZhernosek) [reply IrConsoleApp/Lecture6/Mentor.cs]: Done

Comment (ashurakova) [reply IrConsoleApp/Lecture6/Mentor.cs]: still relevant - use overriding

**Pull request merged**